metasv/age_parser.py

```diff
@@ -73,39 +81,101 @@ def __init__(self, age_out_file=None,tr_region_1=[]):
         self.tr_region_1=tr_region_1 # Truncation region in first sequence
         self.invalid_tr_intervals=[] #Indicies of invalid intervals in first sequence due to truncation
         
-        if age_out_file is None:
-            return
-
-        with open(age_out_file) as age_fd:
+        if age_out_file is not None:
+            self.read_from_age_file(age_out_file)
+
+    class LineReader:
+        """Behaves like file objects do but also counts lines for tracing the AGE parser."""
+        def __init__(self, fd):
+            self.fd = fd
+            self.line_num = 0
+        def readline(self):
+            line = self.fd.readline()
+            if line: self.line_num += 1
+            return line
+
+    rx_rng = re.compile(r"\[\s*(\d+),\s*(\d+)\]")
+    rx_perc = re.compile(r"\(\s*(\d+)%\)")
+    rx_input = re.compile(r"=>\s+(\d+) nucs '(.*?)'")
+    rx_ex = re.compile(r"seq =>\s+(\d+) nucs( \[(\d+),(\d+)\])?")
+    rx_ident = re.compile(r"seq =>\s+(\d+) nucs( \[(\d+),(\d+)\] to \[(\d+),(\d+)\])?")
+
+    def read_alignment_ranges(self, age_fd, name):
+        line = age_fd.readline().strip()
+        if not line.startswith(name):
+            raise AgeFormatError("ALIGNMENT RANGES", age_fd.line_num)
+        start_end = []
+        for m in self.rx_rng.finditer(line):
+            start_end.append([int(m.group(1)), int(m.group(2))])
+        return start_end
+
+    def parse_input_descriptor(self, line):
+        """Returns a tuple of filename and sequence length."""
+        m = self.rx_input.search(line)
+        if m is None:
+            raise AgeFormatError("INPUT DESCRIPTOR", age_fd.line_num)
+        return (m.group(2), int(m.group(1)))
+
```

Comment:
I started testing this and got an error because "age_fd is not defined in parse_input_descriptor"